core/GlobalState.h

```diff
@@ -229,6 +244,21 @@ class GlobalState final {
     UnorderedMap<NameRef, std::string> dslPlugins;
     bool wasModified_ = false;
 
+    // In LSP mode: Used to linearize operations involving lastCommittedLSPEpoch.
+    std::shared_ptr<absl::Mutex> epochMutex;
```

Comment:
can the pointer be `cost`?

Reply:
I don't think so? We assign it in `deepCopy`, so after construction.

Reply:
Yes; I forgot that `deepCopy` calls the constructor, so I can make it an optional constructor argument.